module/VuFind/src/VuFind/Recommend/SearchObjectDeferred.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+/**
+ * Abstract SearchObjectDeferred Recommendations Module (needs to be extended to use
+ * a particular search object).
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Villanova University 2023.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Recommendations
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:recommendation_modules Wiki
+ */
+
+namespace VuFind\Recommend;
+
+/**
+ * Abstract SearchObjectDeferred Recommendations Module (needs to be extended to use
+ * a particular search object).
+ *
+ * This class sets up an AJAX call to trigger a call to some SearchObject implementation.
+ *
+ * @category VuFind
+ * @package  Recommendations
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:recommendation_modules Wiki
+ */
+abstract class SearchObjectDeferred implements RecommendInterface
```

Comment:
I would normally name this AbstractSearchObjectDeferred to indicate that it's an abstract class, but I see that you're following the pattern of the existing SearchObject class, which is also abstract without having Abstract in the name. Maybe we should fix that in a separate PR -- we could rename SearchObject to AbstractSearchObject, then create a deprecated SearchObject class that extends it for backward compatibility purposes. Or maybe it's not worth the effort. Any thoughts/opinions? I like using the Abstract prefix since it groups the abstract classes together and makes it clear that you can't instantiate them... but maybe I'm overrating the value vs. effort.

Reply:
Right I would always use Abstract in the name, but thought following the pattern was more important here.  I'd say consider separately.

Reply:
If we're going to consider this separately, maybe we should go ahead and rename the new class to Abstract now so we don't have to make more changes later. I'm happy to open a PR to do the other renaming in parallel with this work if that would help.